.github/workflows/unit_test.yml

```diff
@@ -42,24 +42,26 @@ jobs:
         echo "KBASE_TEST_TOKEN=${{ secrets.NARRATIVE_TEST_TOKEN }}" >> $GITHUB_ENV
 
     - name: Run Narrative Backend Tests
+      id: test_backend
       shell: bash -l {0}
       run: make test-backend
+      continue-on-error: true
 
     - name: Run Narrative Frontend Unit Tests
+      id: test_frontend
       shell: bash -l {0}
       run: make test-frontend-unit
-
-    - name: make test output available as artifact in case of failure
-      if: ${{ failure() }}
-      uses: actions/upload-artifact@v2
-      with:
-        name: karma-result.json
-        path: karma-result.json
+      continue-on-error: true
 
     - name: Send to Codecov
       uses: codecov/codecov-action@v1.5.2
+      continue-on-error: true
       with:
         file: |
           ./python-coverage/coverage.xml
           ./js-coverage/lcov/lcov.info
         fail_ci_if_error: true
+
+    - name: outcome
```

Comment:
assess whether the backend or frontend tests failed, and exit with an error if so

Reply:
The sending to codecov should also get this treatment. If tests failed, then no coverage to send, and that'll bomb in annoying ways.

Reply:
Fixed, thanks for the suggestion!